pkg/distributor/ha_tracker.go

```diff
@@ -212,10 +212,47 @@ func newHATracker(cfg HATrackerConfig, limits haTrackerLimits, reg prometheus.Re
 		t.client = client
 	}
 
-	t.Service = services.NewBasicService(nil, t.loop, nil)
+	t.Service = services.NewBasicService(t.syncHATrackerStateOnStart, t.loop, nil)
 	return t, nil
 }
 
+func (h *haTracker) syncHATrackerStateOnStart(ctx context.Context) error {
+	if !h.cfg.EnableHATracker {
+		return nil
+	}
+	// haTracker holds HATrackerConfig with the prefix.
+	keys, err := h.client.List(ctx, "")
+	if err != nil {
+		return err
+	}
+
+	if len(keys) == 0 {
+		level.Warn(h.logger).Log("msg", "sync HA state on start: no keys for HA tracker prefix", "err", err)
+		return nil
+	}
+
+	for i := 0; i < len(keys); i++ {
+		if ctx.Err() != nil {
+			return fmt.Errorf("syncing HA tracker state on startup: %w", context.Cause(ctx))
+		}
+
+		val, err := h.client.Get(ctx, keys[i])
+		if err != nil {
+			level.Warn(h.logger).Log("msg", "sync HA state on start: failed to get replica value", "key", keys[i], "err", err)
+			return err
+		}
+
+		desc, ok := val.(*ReplicaDesc)
+		if !ok {
+			level.Error(h.logger).Log("msg", "sync HA state on start: got invalid replica descriptor", "key", keys[i])
```

Comment:
again on adding the implications to the log line. This is less clear, but maybe mentioning that the value in the key is skipped should be enough.

Reply:
Yeah makes sense. I have updated it at [253c56a](https://github.com/grafana/mimir/pull/9826/commits/253c56a2f0ce5cc09dcb48cc1eec2273eabdcf3a). Please let me know what you. think